airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -70,11 +70,17 @@ def records_with_state(records, state, stream_mapping, state_cursor_paths) -> It
         cursor_field = helper.field(stream.cursor_field)
         record_value = cursor_field.parse(record=record.record.data)
         try:
+            if state[stream_name] is None:
+                continue
+
             # first attempt to parse the state value assuming the state object is namespaced on stream names
             state_value = cursor_field.parse(record=state[stream_name], path=state_cursor_paths[stream_name])
         except KeyError:
-            # try second time as an absolute path in state file (i.e. bookmarks -> stream_name -> column -> value)
-            state_value = cursor_field.parse(record=state, path=state_cursor_paths[stream_name])
+            try:
+                # try second time as an absolute path in state file (i.e. bookmarks -> stream_name -> column -> value)
+                state_value = cursor_field.parse(record=state, path=state_cursor_paths[stream_name])
+            except KeyError:
```

Comment:
@brianjlai do you know why this sometimes fails to find the state value?

Reply:
So I am not sure this was working as intended up to this point. In the original comparison, we tested all the `record` messages against the latest `state` message. This works for the latest state message because it carries the final states of all prior streams. 

But when we compare each stream one at a time forward, the current `state` message might not have state for the upcoming stream if it is the first one. This allows us to basically skip cursor comparisons if there is no cursor to compare on in state. Also, when reading w/ state, we get messages for the remaining streams in the configured catalog which we should also skip since there is no cursor in the state for it yet. And we'll check it on subsequent reads

Reply:
got it. makes sense!